cmds/cksum/cksum.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+func helpPrinter() {
+	fmt.Printf("Usage:\ncksum <File Name>\n")
+	flag.PrintDefaults()
+	os.Exit(0)
+}
+
+func versionPrinter() {
+	fmt.Println("cksum utility, URoot Version.")
+	os.Exit(0)
+}
+
+func GetInput(fileName string) (input []byte, err error) {
+	if fileName != "" {
+		return ioutil.ReadFile(fileName)
+	}
+	return ioutil.ReadAll(os.Stdin)
+}
+
+func printCksum(input []byte) string {
+
+	var cksumTable = []uint64{
+		/*
+		 * Following table is originally contributed by
+		 * Q. Frank Xia et. al (qx@math.columbia.edu)
+		 * under GPL License.
+		 * http://ftp-archive.freebsd.org/pub/FreeBSD-Archive/
+		 * old-releases/i386/1.0-RELEASE/ports/textutils/src/cksum.c
```

Comment:
Have you tried using https://golang.org/pkg/hash/crc32/ ?
Are they different? Sorry if this is a dumb question

Reply:
Thanks for the those input.
I had tried hash/crc32 earlier: [Link](https://github.com/MishraShivendra/u-root/commit/410e2e90e504b414b8b1a28ac213f856507a7ef7#diff-dd91284fa77f7a5577aa054eca62bdf2L33) .
But for some reason, It returns a different value than the Linux cksum utility (given same data and polynomial).
Current changes seems easier to me, so used.